elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/GraphQLDSL.java

```diff
@@ -0,0 +1,585 @@
+            TypedOperation.OperationType operationType,
+            SelectionSet selectionSet
+    ) {
+        return new TypedOperation(operationType, null, null, selectionSet);
+    }
+
+    /**
+     * Creates a typed query definition
+     *
+     * @param operationType  The type of the query definition.
+     * @param name  An unique identifier of a {@link OperationDefinition operation} in a multi-operations
+     * {@link Document GraphQL query}
+     * @param variableDefinitions  A collection of GraphQL variable definitions
+     * @param selectionSet  The definition/selection-spec of a multi-definitions GraphQL query for
+     *
+     * @return a definition of a multi-definitions GraphQL query
+     *
+     * @see <a href="https://graphql.org/learn/schema/#the-query-and-mutation-types">Typed Query</a>
+     * @see <a href="https://graphql.org/learn/queries/#variables">Variables</a>
+     */
+    public static TypedOperation typedOperation(
+            TypedOperation.OperationType operationType,
+            String name,
+            VariableDefinitions variableDefinitions,
+            SelectionSet selectionSet
+    ) {
+        return new TypedOperation(operationType, name, variableDefinitions, selectionSet);
+    }
+
+    /**
+     * Creates a collection of GraphQL variable definitions.
+     *
+     * @param variableDefinitions  A variable-sized array of {@link VariableDefinition}s
+     *
+     * @return A collection of GraphQL variable definitions
+     *
+     * @see <a href="https://graphql.org/learn/queries/#variables">Variables</a>
+     */
+    public static VariableDefinitions variableDefinitions(VariableDefinition... variableDefinitions) {
+        return new VariableDefinitions(Arrays.asList(variableDefinitions));
+    }
+
+    /**
+     * Creates a single GraphQL variable definition without default value.
+     *
+     * @param variable  A string (without "$" sign) representing the name of the variable
+     * @param type  A simplified String representation of the aforementioned type literal.
+     *
+     * @return A single GraphQL variable definition
+     *
+     * @see <a href="https://graphql.org/learn/queries/#variables">Variables</a>
+     */
+    public static VariableDefinition variableDefinition(String variable, String type) {
+        return new VariableDefinition(variable, type, null);
+    }
+
+    /**
+     * Creates a single top-level entity(object field) selection without {@link Argument}s.
+     *
+     * @param name  The name of the selected entity/field that would appear in a GraphQL query
+     * @param selectionSet  The fields of the entity that are selected
+     *
+     * @return a top-level selection
+     *
+     * @see <a href="https://graphql.org/learn/queries/#fields">Fields</a>
+     * @see <a href="https://graphql.org/learn/schema/#object-types-and-fields">Object Types and Fields</a>
+     */
+    public static Selection entity(String name, SelectionSet selectionSet) {
```

Comment:
Let's collapse entity and field into a single concept - and call it field.   Fewer concepts == better.

Reply:
Done